lib/site_setting_extension.rb

```diff
@@ -695,9 +699,9 @@ def setting(name_arg, default = nil, opts = {})
 
       if opts[:area]
         split_areas = opts[:area].split("|")
-        if split_areas.any? { |area| !SiteSetting::VALID_AREAS.include?(area) }
+        if split_areas.any? { |area| !SiteSetting.valid_areas.include?(area) }
           raise Discourse::InvalidParameters.new(
-                  "Area is incorrect. Valid areas: #{SiteSetting::VALID_AREAS.join(", ")}",
+                  "Area is incorrect. Valid areas: #{SiteSetting.valid_areas.join(", ")}",
```

Comment:
Minor but I would change this error message slightly

```suggestion
                  "Area is invalid, valid areas are: #{SiteSetting.valid_areas.join(", ")}",
```